src/main.rs

```diff
@@ -20,29 +51,176 @@ pub struct ParseError {
+                    .collect();
+            }
+
+            Rule::byminute_expr => {
+                rrule_result.by_minute = line
+                    .into_inner()
+                    .next()
+                    .unwrap()
+                    .as_str()
+                    .split(",")
+                    .collect();
+            }
+
+            Rule::bysecond_expr => {
+                rrule_result.by_second = line
+                    .into_inner()
+                    .next()
+                    .unwrap()
+                    .as_str()
+                    .split(",")
+                    .collect();
             }
 
+            Rule::byday_expr => {
+                rrule_result.by_day = line
+                    .into_inner()
+                    .next()
+                    .unwrap()
+                    .as_str()
+                    .split(",")
+                    .collect();
+            }
+
+            Rule::bymonthday_expr => {
+                rrule_result.by_month_day = line
+                    .into_inner()
+                    .next()
+                    .unwrap()
+                    .as_str()
+                    .split(",")
+                    .collect();
+            }
+
+            Rule::byyearday_expr => {
+                rrule_result.by_year_day = line
+                    .into_inner()
+                    .next()
+                    .unwrap()
+                    .as_str()
+                    .split(",")
+                    .collect();
+            }
             _ => {}
         }
     }
 }
 
+// ToDo : Add validation for checking that the RRULE string was properly extracted from the parser
+// by counting ';' in the original rrule string and ':' in the parsed json
 fn main() {
     let args: Vec<String> = env::args().collect();
-    let s = "FREQ=YEARLY;COUNT=6;BYDAY=TU,TH".to_owned();
-    convert_rrule_to_json(s)
+    //    let s = "FREQ=MONTHLY;INTERVAL=1;BYHOUR=9;BYMINUTE=1;BYMONTHDAY=15,27".to_owned();
+
+    let s = "FREQ=DAILY".to_owned();
+    let mut rrule_result = RRule {
+        frequency: String::from(""),
+        count: String::from(""),
+        interval: String::from(""),
+        by_hour: Vec::new(),
+        by_minute: Vec::new(),
+        by_second: Vec::new(),
+        by_day: Vec::new(),
+        by_month_day: Vec::new(),
+        by_year_day: Vec::new(),
+    };
+
+    convert_to_rrule(&mut rrule_result, &s);
+    println!("Json is {}", rrule_result.to_json());
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::{convert_to_rrule, RRule};
+    use serde_json::json;
+
+    struct RRuleTestCase<'a> {
+        rrule_string: &'a str,
+        expected_flat_json: &'a str,
+    }
+
+    #[test]
+    fn test_we_can_parse_to_proper_json() {
+        let rrule_test_cases: Vec<RRuleTestCase> = vec![
+            RRuleTestCase {
+                rrule_string: "FREQ=DAILY",
+                expected_flat_json: r#"{"frequency":"DAILY"}"#,
+            },
+            RRuleTestCase {
```

Comment:
you could write this as a macro. but that is for another day... 💃

Reply:
thanks! will try to incorporate in the next PR